packages/magento-product-configurable/components/ConfigurableProductUrls/ConfigurableProductUrls.tsx

```diff
@@ -0,0 +1,40 @@
+import {
+  ProductLinkProps,
+  useProductLink,
+} from '@graphcommerce/magento-product/hooks/useProductLink'
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
+import { ConfigurableOptionsFragment } from '../../graphql/ConfigurableOptions.gql'
+import { useConfigurableOptionsSelection } from '../../hooks/useConfigurableOptionsSelection'
+
+type ConfigurableProductUrlsProps = {
+  index?: number
+  product: Partial<ConfigurableOptionsFragment>
+}
+
+export function ConfigurableProductUrls(props: ConfigurableProductUrlsProps) {
+  const { product, index = 0 } = props
+  const router = useRouter()
+  const variant = useConfigurableOptionsSelection({ url_key: product?.url_key, index }).configured
+    ?.configurable_product_options_selection?.variant
+  const urlKey = variant?.url_key ?? product?.url_key
+
+  const productLink = useProductLink({
+    url_key: urlKey,
+    __typename: variant?.__typename as ProductLinkProps['__typename'],
+  })
```

Comment:
I think it is better to not use useProductLink, but use productLink directly then it allows you to write something like this directly inside the useEffect:

```
const link = variant?.url_key ? productLink(variant) : productLink(product)
```